### PR TITLE
docs: explain `migrate` CLI magic in reference

### LINKED ISSUE #3600: Explain `migrate` 'magic' in the CLI reference
Some of the `migrate` commands do extra 'magic' which is not very well explained in the CLI reference. For example the [`migrate reset` introduction](https://www.prisma.io/docs/reference/api-reference/command-reference#migrate-reset) says 'This command deletes and recreates the database, or performs a 'soft reset' by removing all data, tables, indexes, and other artifacts.', but it also:
- applies the migrations you have in your migrations file afterwards
- regenerates the Prisma Client
- reseeds the database if there is a seed script

This extra information should be in the overview for each command.

### PR DESCRIPTION
**Pull request by ruheni**
Closes #3600

Add more information to the `migrate dev` and `migrate reset` commands in the reference

### CODE REVIEW

File: content/400-reference/200-api-reference/200-command-reference.mdx [hunk 1]
```diff
@@ -1033,7 +1039,12 @@ prisma migrate dev --create-only
 
 **For use in development environments only**
 
-This command deletes and recreates the database, or performs a 'soft reset' by removing all data, tables, indexes, and other artifacts.
+This command:
+
+1. Drops the database/schema¹ if possible, or performs a soft reset if the environment does not allow deleting databases/schemas¹
```
Comment (jharrell) [inline content/400-reference/200-api-reference/200-command-reference.mdx]: What are these footnotes referencing? That's not clear to me.

Comment (ruheni) [reply content/400-reference/200-api-reference/200-command-reference.mdx]: Copy-pasta mistake. I'll remove it 😄

Comment (ruheni) [inline content/400-reference/200-api-reference/200-command-reference.mdx]:
```suggestion
1. Drops the database/schema if possible, or performs a soft reset if the environment does not allow deleting databases/schemas
```

File: content/400-reference/200-api-reference/200-command-reference.mdx [hunk 2]
```diff
@@ -1033,7 +1039,12 @@ prisma migrate dev --create-only
 
 **For use in development environments only**
 
-This command deletes and recreates the database, or performs a 'soft reset' by removing all data, tables, indexes, and other artifacts.
+This command:
+
+1. Drops the database/schema if possible, or performs a soft reset if the environment does not allow deleting databases/schemas
+1. Creates a new database/schema¹ with the same name if the database/schema¹ was dropped
```
Comment (ruheni) [inline content/400-reference/200-api-reference/200-command-reference.mdx]:
```suggestion
1. Creates a new database/schema with the same name if the database/schema¹ was dropped
```

File: content/400-reference/200-api-reference/200-command-reference.mdx [hunk 3]
```diff
@@ -1033,7 +1039,12 @@ prisma migrate dev --create-only
 
 **For use in development environments only**
 
-This command deletes and recreates the database, or performs a 'soft reset' by removing all data, tables, indexes, and other artifacts.
+This command:
+
+1. Drops the database/schema if possible, or performs a soft reset if the environment does not allow deleting databases/schemas
+1. Creates a new database/schema with the same name if the database/schema¹ was dropped
```
Comment (ruheni) [inline content/400-reference/200-api-reference/200-command-reference.mdx]:
```suggestion
1. Creates a new database/schema with the same name if the database/schema was dropped
```

**Review by jharrell — APPROVED** [review]

**Comment by janpio** [comment]
This PR makes the documentation of these two commands worse. 

The level of details that is now the default does not matter to most users - the previous sentence really just described the outcome. The new information on the other hand includes details that are good to be mentioned _somewhere_, but not as a replacement for the fundamental information.

I suggest we restore the previous information, and somehow position this more details rundown below it.

File: content/400-reference/200-api-reference/200-command-reference.mdx [hunk 4]
```diff
@@ -980,7 +980,13 @@ Instead of `migrate dev` and related commands, [`db push`](/concepts/components/
 
 **For use in development environments only, requires shadow database**
 
-The `migrate dev` command updates your database using migrations during development and creates the database if it does not exist.
+The `migrate dev` command:
+
+1. Reruns the existing migration history in the [shadow database](/concepts/components/prisma-migrate/shadow-database) in order to detect schema drift (edited or deleted migration file, or a manual changes to the database schema)
+1. Applies pending migrations to the shadow database (for example, new migrations created by colleagues)
```
Comment (janpio) [inline content/400-reference/200-api-reference/200-command-reference.mdx:986]: Are you sure these are only applied to the shadow database?

**Pull request merged**